pkg/graveler/graveler.go

```diff
@@ -784,20 +787,26 @@ func (id TagID) String() string {
 }
 
 type KVGraveler struct {
-	db             *DBGraveler
-	hooks          HooksHandler
-	RefManager     RefManager
-	StagingManager StagingManager
-	log            logging.Logger
+	db                       *DBGraveler
+	hooks                    HooksHandler
+	CommittedManager         CommittedManager
+	RefManager               RefManager
+	StagingManager           StagingManager
+	branchLocker             BranchLocker
+	protectedBranchesManager ProtectedBranchesManager
+	log                      logging.Logger
 }
 
 func NewKVGraveler(branchLocker BranchLocker, committedManager CommittedManager, stagingManager StagingManager, refManager RefManager, gcManager GarbageCollectionManager, protectedBranchesManager ProtectedBranchesManager) *KVGraveler {
 	return &KVGraveler{
-		db:             NewDBGraveler(branchLocker, committedManager, stagingManager, refManager, gcManager, protectedBranchesManager),
-		hooks:          &HooksNoOp{},
-		RefManager:     refManager,
-		StagingManager: stagingManager,
-		log:            logging.Default().WithField("service_name", "graveler_graveler"),
+		db:                       NewDBGraveler(branchLocker, committedManager, stagingManager, refManager, gcManager, protectedBranchesManager),
+		hooks:                    &HooksNoOp{},
+		CommittedManager:         committedManager,
+		RefManager:               refManager,
+		StagingManager:           stagingManager,
+		branchLocker:             branchLocker,
```

Comment:
No need to add branchLocker to the KV implementation. You can just use g.db.branchLcoker

Reply:
done